CONTRIBUTING.md

```diff
@@ -77,6 +78,60 @@ to do this is to invoke `rustc` like this:
 $ RUST_BACKTRACE=1 rustc ...
 ```
 
+## The Build System
+
+The build system for Rust is complex. It covers bootstrapping the compiler,
+running tests, building documentation and more. Unless you are familiar with
+Makefiles, I wouldn't suggest trying to understand everything going on in
```

Comment:
What would you suggest as an alternative? Do you think it would be OK to cut that bit off, and leave it as

> ...advanced Make features which can make it difficult to understand.

?